src/js/phet-converter.js

```diff
@@ -0,0 +1,181 @@
+    addressTemplate: "/%phetioID/%eventType/%event",
+
+    jsToOSCTypes: {
+        "number": "d",
+        "string": "s"
+    },
+
+    invokers: {
+        toOSC: "phetosc.converter.toOSC({arguments}.0, {that})"
+    },
+
+    components: {
+        filter: {
+            type: "phetosc.eventFilter"
+        }
+    },
+
+    events: {
+        onError: null
+    },
+
+    listeners: {
+        "onError.logWarningMessage": {
+            funcName: "fluid.log",
+            args: [
+                fluid.logLevel.WARN,
+                "{arguments}.0", // Error message.
+                " Address: ",
+                "{arguments}.1",
+                " Value: ",
+                "{arguments}.2"
+            ]
+        }
+    }
+});
+
+phetosc.converter.oscTypeForJSType = function (jsType, jsToOSCTypes) {
+    return jsToOSCTypes[jsType];
+};
+
+phetosc.converter.toMessage = function (address, that, val, paramType) {
+    var mappedAddress = that.options.addressMap[address];
+    if (mappedAddress) {
+        // Just in case users forget the OSC-required leading slash.
+        address = mappedAddress.indexOf("/") !== 0 ?
+            "/" + mappedAddress : mappedAddress;
+    }
+
+    var message = {
+        address: address
+    };
+
+    var arg = val === undefined ? undefined :
+        paramType === "boolean" ? phetosc.converter.toBooleanMessage(val) :
+        phetosc.converter.toArgumentMessage(address, that, val, paramType);
+
+    if (arg) {
+        message.args = [arg];
+    }
+
+    return [message];
+};
+
+phetosc.converter.toBooleanMessage = function (val) {
+    return {
+        type: val ? "T" : "F"
+    };
+};
+
+phetosc.converter.toArgumentMessage = function (address, that, val, paramType) {
+    var oscType = phetosc.converter.oscTypeForJSType(paramType,
+        that.options.jsToOSCTypes);
+
+    if (!oscType) {
+        that.events.onError.fire("A parameter property was found with a type of " +
+            paramType + ", which is not OSC compatible. It was omitted.",
+            address, val);
+
+        return;
+    }
+
+    return {
+        type: oscType,
+        value: val
+    };
+};
+
+phetosc.converter.parameterToMessages = function (addressPrefix, that, val, key) {
+    if (fluid.isArrayable(val)) {
+        that.events.onError.fire("An Array-typed event parameter was found; conversion of Array parameters to OSC messages is not yet supported. It was omitted.",
+            val, key);
+
+        return [];
+    }
+
+    if (that.options.excludeParameters.indexOf(key) > -1) {
+        // Skip the parameter.
+        return [];
+    }
+
+    var address = addressPrefix + "/" + key,
+        paramType = typeof val;
+
+    return paramType === "object" ?
+        phetosc.converter.objectToMessages(address, that, val) :
+        phetosc.converter.toMessage(address, that, val, paramType);
+};
+
+phetosc.converter.objectToMessages = function (addressPrefix, that, obj) {
+    if (!obj || Object.keys(obj) < 1) {
```

Comment:
Sorry, ignore 2nd half of comment, I just scrolled down a little further on this tiny screen. Function body might be a touch clearer with 2nd half enclosed in "else"